plugins/web/opentelemetry-instrumentation-user-interaction/src/instrumentation.ts

```diff
@@ -464,41 +489,54 @@ export class UserInteractionInstrumentation extends InstrumentationBase<unknown>
           Array.isArray(applyArgs) && applyArgs[0] instanceof Event
             ? applyArgs[0]
             : undefined;
-        const target = event?.target;
+
+        const target = plugin._allowEventType(task.eventName) && event?.target;
         let span: api.Span | undefined;
-        const activeZone = this;
+
         if (target) {
-          span = plugin._createSpan(target, task.eventName);
+          if (!plugin._currentInteractionZone) {
+            // create new span for interaction from current task zone
+            span = plugin._createSpan(target, task.eventName);
+
+            if (span) {
+              api.context.with(
+                api.trace.setSpan(api.context.active(), span),
+                () => {
+                  plugin._currentInteractionZone = Zone.current;
+                }
+              );
```

Comment:
Why is this required to be run in a context? Does it have something to do with `Zone.current`?

Reply:
Your question is valid, it doesn't really need to run in context. I believe I did this to create a separate Zone from the listener task's Zone since the goal here is to share a Zone across multiple tasks that don't may not have common parent. 

At the time it felt semantically off to state the first registered listener's zone should be considered the zone for all other registered listeners. But in reality, it doesn't really matter since this execution context is never really used.